### PR TITLE
Feature/simple sensor simulator unit test

### PR DESCRIPTION
**Pull request by SzymonParapura**
# Description

## Abstract

This PR contains unit tests for:
1. Box
2. Raycaster
3. Vertex 
4. GridTraversal

## Details

List of tests included in this PR:

1. toVertex_onePoint
2. toVertex_manyPoints
3. toVertex_empty
4. toPoint_oneVertex
5. toPoints_manyVertices
6. toPoints_empty
7. Box
8. begin
9. end
10. iterator_operator_dereference
11. iterator_operator_increment
12. iterator_operator_notEqual
13. addPrimitive_box
14. addPrimitive_twoIdenticalNames
15. raycast_empty
16. raycast_box
17. setDirection_oneBox
18. setDirection_manyBoxes
19.  getDetectedObjects

## References

Jira ticket: [link](https://tier4.atlassian.net/browse/RJD-937)

# Destructive Changes

There are no destructive changes

### CODE REVIEW

**Comment by SzymonParapura** [comment]
I have fixed a spelling mistake. Now this PR is ready for review.

File: simulation/simple_sensor_simulator/CMakeLists.txt [hunk 1]
```diff
@@ -84,7 +84,9 @@ EXPORT export_simple_sensor_simulator_component
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(ament_cmake_gtest REQUIRED)
```
[the remark this answers is not in the document]
Comment (SzymonParapura) [reply simulation/simple_sensor_simulator/CMakeLists.txt:87]: Ok, I will do

[the remark this answers is not in the document]
Comment (SzymonParapura) [reply simulation/simple_sensor_simulator/CMakeLists.txt]: It is done

[the remark this answers is not in the document]
Comment (HansRobo) [reply simulation/simple_sensor_simulator/CMakeLists.txt]: Thank you!

**Review by HansRobo — APPROVED** [review]
LGTM!

**Pull request merged**